fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/query/FDBAndQueryToIntersectionTest.java

```diff
@@ -105,13 +106,17 @@ public void testComplexQueryAndWithTwoChildren(boolean shouldDeferFetch) throws
                     coveringIndexScan(indexScan(allOf(indexName("MySimpleRecord$str_value_indexed"), bounds(hasTupleString("[[even],[even]]"))))),
                     coveringIndexScan(indexScan(allOf(indexName("MySimpleRecord$num_value_3_indexed"), bounds(hasTupleString("[[3],[3]]"))))),
                     equalTo(field("rec_no")))));
-            assertEquals(-929788310, plan.planHash());
+            assertEquals(-929788310, plan.planHash(PlanHashable.PlanHashKind.LEGACY));
```

Comment:
This is where my PR and your PR will clash like crazy as the new planner already creates different hashes.

Reply:
Based on offline discussion - here is what we're shooting for:
1. Add a method
`protected void assertPlanHash(QueryPlanner planner, RecordQueryPlan plan, PlanHashable.PlanHashKind kind, int expectedOriginalHash, int expectedCascadesHash)  `
to the test base class
2. Convert each `@DualPlannerTest`  assert block from
```
        assertEquals(-447322749, plan2.planHash(PlanHashable.PlanHashKind.LEGACY));
        assertEquals(-40431985, plan2.planHash(PlanHashable.PlanHashKind.FOR_CONTINUATION));
        assertEquals(453584239, plan2.planHash(PlanHashable.PlanHashKind.STRUCTURAL_WITHOUT_LITERALS));
```
to
```
        assertPlanHash(planner, plan2, PlanHashable.PlanHashKind.LEGACY, -447322749, -447322749);
        assertPlanHash(planner, plan2, PlanHashable.PlanHashKind.FOR_CONTINUATION, -40431985, -40431985);
        assertPlanHash(planner, plan2, PlanHashable.PlanHashKind.STRUCTURAL_WITHOUT_LITERALS, 453584239, 453584239);
```

@normen662 will implement as part of the other PR